test/snippet/io/alignment_file/alignment_file_input_my_traits.cpp

```diff
@@ -0,0 +1,17 @@
+#include <seqan3/alphabet/nucleotide/dna4.hpp>
+#include <seqan3/io/alignment_file/input.hpp>
+#include <seqan3/range/container/bitcompressed_vector.hpp>
+
+struct my_traits : seqan3::alignment_file_input_default_traits<>
+{
+    using sequence_alphabet = seqan3::dna4;                        // instead of dna5
+
+    template <typename alph>
+    using sequence_container = seqan3::bitcompressed_vector<alph>; // must be defined as a template!
+};
+
+// ... within main you can then use:
+int main()
+{
+    seqan3::alignment_file_input<my_traits> fin{"/tmp/my.sam"};
```

Comment:
now this snippet cannot run by copy'n'paste since no one ever created the file. Use a string stream as well.